### PR TITLE
Explain that rolling update is not supported between feature updates …

### PR DESCRIPTION
**Pull request by rymsha**
…#309

### CODE REVIEW

File: docs/release/upgrade.adoc [hunk 1]
```diff
@@ -22,7 +22,7 @@ We recommend the following procedure:
 . Upgrade all other nodes. If you run a combination of back-end and front-end nodes, we recommend upgrading the back-end nodes before you move on to the front-end nodes.
 
 If this is done correctly, bugfix upgrades (v7.x.y) can be executed without downtime.
-For feature upgrades (7.x.0), the forming of the cluster is a little bit more complicated, and there may be some requests to the site that are rejected during the upgrade but the process should still be really smooth.
+For feature upgrades (7.x.0), the forming of the cluster is more complicated and require full cluster restart. Rolling update is sometimes possible but not recommended/supported out of the box.
```
Comment (gbbirkisson) [inline docs/release/upgrade.adoc:25]: I would say we recommend rolling updates all the time to our customers. Shouldn't we rephrase this to say that in some cases it is not supported.

Comment (rymsha) [reply docs/release/upgrade.adoc:25]: Problem with rolling update is that it actually may not work as expected.

Simple (not real) example:

version 7.8.0 introduces new internal filed "_newFiled" to node data.
version 7.7.0 does not know about such filed.

During migration node with 7.8.0 version writes a new node and sets "_newFiled".
Same cluster, but different node with 7.7.0 version updates that nearly created node, and because "_newField" is not known - erases anything written to that filed.


It is just one example. Many other complications can appear - due to the fact that we care about "backwards compatibility", but we don't do anything in specific to support "forward compatibility" - old versions don't know how future versions will manipulate the data/change communication protocols.

Comment (sigdestad) [reply docs/release/upgrade.adoc:25]: @rymsha - afaik we will know if a feature release will require a full cluster release or not, and could specify this in upgrade notes for the version, right?

And maybe rephrase the section to something like: "we generally recommend full cluster restart for feature upgrades, unless documentation clearly states this upgrade can be performed as a rolling upgrade" ish

Comment (rymsha) [reply docs/release/upgrade.adoc:25]: In general it is hard to know if we break forward compatibility. People should expect the unexpected, anyway.
We may know that it is certainly broken in some cases, like in 7.7.0

**Review by alansemenov — APPROVED** [review]
